Source/JavaWrappers.cpp

```diff
@@ -274,6 +282,116 @@ namespace java::lang
     }
 }
 
+namespace java::websocket
+{
+    jclass WebSocketClient::g_webSocketClass{};
+    std::vector<std::pair<jobject, WebSocketClient*>> WebSocketClient::s_instances;
+
+    WebSocketClient::WebSocketClient(std::string url, std::function<void()> open_callback, std::function<void()> close_callback, std::function<void(std::string)> message_callback, std::function<void()> error_callback)
+        : Object{g_webSocketClass}
+        , m_openCallback{std::move(open_callback)}
+        , m_messageCallback{std::move(message_callback)}
+        , m_closeCallback{std::move(close_callback)}
+        , m_errorCallback{std::move(error_callback)}
+    {
+        static JNINativeMethod methods[] =
+        {
+            {"closeCallback", "()V", (void*)OnClose},
+            {"openCallback", "()V", (void*)OnOpen},
+            {"messageCallback", "(Ljava/lang/String;)V", (void*)OnMessage},
+            {"errorCallback", "()V", (void*)OnError},
+        };
+        m_env->RegisterNatives(m_class, methods, 4);
+
+        JObject(m_env->NewObject(m_class, m_env->GetMethodID(m_class, "<init>", "(Ljava/lang/String;)V"), m_env->NewStringUTF(url.c_str())));
+
+        s_instances.push_back(std::make_pair(JObject(), this));
+
+        jmethodID connectSocket{m_env->GetMethodID(m_class, "connectBlocking", "()Z")};
+        m_env->CallBooleanMethod(JObject(), connectSocket);
+    }
+
+    WebSocketClient::~WebSocketClient()
+    {
+        s_instances.erase(std::remove_if(s_instances.begin(), s_instances.end(), [this](const auto& p) { return p.second == this; }), s_instances.end());
+    }
+
+    void WebSocketClient::OnOpen(JNIEnv* env, jobject obj) 
+    {
+        auto itObject = FindInstance(env, obj);
+        if (itObject != nullptr)
+        {
+            itObject->m_openCallback();
+        }
+    }
+
+    void WebSocketClient::OnMessage(JNIEnv* env, jobject obj, jstring message) 
+    {
+        auto itObject = FindInstance(env, obj);
+        if (itObject != nullptr)
+        {
+            java::lang::String mystr{message};
+            itObject->m_messageCallback(mystr);
+        }
+    }
+
+    void WebSocketClient::OnClose(JNIEnv* env, jobject obj) 
+    {
+        auto itObject = FindInstance(env, obj);
+        if (itObject != nullptr)
+        {
+            itObject->m_closeCallback();
+        }
+    }
+
+    void WebSocketClient::OnError(JNIEnv* env, jobject obj) 
+    {
+        auto itObject = FindInstance(env, obj);
+        if (itObject != nullptr)
+        {
+            itObject->m_errorCallback();
+        }
+    }
+
+    java::websocket::WebSocketClient* WebSocketClient::FindInstance(JNIEnv* env, jobject obj)
+    {
+        const auto it = std::find_if(s_instances.begin(), s_instances.end(), [&obj, &env](const auto &it)
+        {
+            return env->IsSameObject(obj, it.first);; // Comparing with the object
```

Comment:
nit: double semicolon and not sure if the comment is adding anything
```suggestion
            return env->IsSameObject(obj, it.first);
```